torch/_dynamo/output_graph.py

```diff
@@ -894,8 +894,7 @@ def call_user_compiler(self, gm: fx.GraphModule) -> CompiledFn:
         torch._dynamo.utils.increment_op_count(tot)
         for pl in placeholders:
             arg = pl.meta["grapharg"]
-            # TODO: Why isn't this stored in meta :think:
-            pl._dynamo_source = arg.source
+            pl.meta["dynamo_source"] = arg.source
```

Comment:
IIRC the answer to this question is two fold (1) avoid leaking dynamo details down to node metas (2) To avoid accidental reliance on this in place that can have non dynamo entry points without an explicit hasattr check. 

It's sort of "not good enough" for meta, if that makes sense? 

That being said, I do not have a strong opinion here.

Reply:
Also, I am not sure where meta is propagated when nodes are copied, you may need manual binding.

Reply:
I did a little grep on "stack_trace" which follows a similar pattern

<img width="431" alt="image" src="https://github.com/pytorch/pytorch/assets/4755252/c30914cd-8883-4973-abb6-6194e3ada156">

Check out where we do that, and see if we can add "dynamo_source" to a spot here.